code_annotations/find_django.py

```diff
@@ -0,0 +1,168 @@
+        for model in self.local_models.union(self.non_local_models):
+            model_id = get_model_id(model)
+            heirarchy = inspect.getmro(model)
+            model_annotations = []
+
+            # See if any annotations exist in the docstring
+            for obj in heirarchy:
+                if obj.__doc__ is not None:
+                    if any(anno in obj.__doc__ for anno in annotation_tokens):
+                        # Read in the source file to get the line number
+                        filename = inspect.getsourcefile(obj)
+                        with open(filename, 'r') as file_handle:
+                            txt = file_handle.read()
+
+                        for inner_match in re.finditer(query, obj.__doc__):
+                            # TODO: This is duplicated code with extensions/base.py
+                            # No matter how long the regex is, there should only be 2 non-None items,
+                            # with the first being the annotation token and the 2nd being the comment.
+                            cleaned_groups = [item for item in inner_match.groups() if item is not None]
+
+                            if len(cleaned_groups) != 2:  # pragma: no cover
+                                raise Exception('{}: Number of found items in the list is not 2. Found: {}'.format(
+                                    get_model_id(obj),
+                                    cleaned_groups
+                                ))
+
+                            annotation, comment = cleaned_groups
+
+                            # Get the line number by counting newlines + 1 (for the first line).
+                            # Note that this is the line number of the beginning of the comment, not the
+                            # annotation token itself. We find based on the entire code content of the model
+                            # as that seems to be the only way to be sure we're getting the correct line number.
+                            # It is slow and should be replaced if we can find a better way that is accurate.
+                            line = txt.count('\n', 0, txt.find(inspect.getsource(obj))) + 1
+
+                            model_annotations.append({
+                                'found_by': "django",
+                                'filename': filename,
+                                'line_number': line,
+                                'annotation_token': annotation.strip(),
+                                'annotation_data': comment.strip(),
+                                'extra': {
+                                    'object_id': model_id,
+                                    'full_comment': obj.__doc__.strip()
+                                }
+                            })
+            if model_annotations:
+                if model_id in safelisted_models:
+                    self._add_error("{} is annotated, but also in the safelist.".format(model_id))
```

Comment:
Does this need to be an error rather than a warning?  Imagine that somebody annotates the model inline, which causes code-annotations to fail CI for all consuming repositories (presumably they were previously remotely annotating that model).

Reply:
I'd thought about that, and I think that failing CI might be the only way they'll know that they can remove it. It should only happen when a library is upgraded, and the person doing the upgrade should be responsible for removing the local annotation, just like they're responsible for migrations. I believe it's the only way we can actually keep them accurate.

Reply:
Oh, right, forgot they have to bump the library version anyway (on the consumer side).